source/common/upstream/cluster_manager_impl.cc

```diff
@@ -468,6 +558,11 @@ bool ClusterManagerImpl::removeCluster(const std::string& cluster_name) {
   if (removed) {
     cm_stats_.cluster_removed_.inc();
     updateGauges();
+    // Did we ever deliver merged updates for this cluster?
+    if (updates_map_.count(cluster_name) == 1) {
+      // No need to manually disable timers, this should take care of it.
+      updates_map_.erase(cluster_name);
```

Comment:
perf nit: Either just universally call erase() or do a find() to return an iterator and then erase that.